src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -268,6 +269,10 @@ public function lists($column, $key = null)
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage < 0) {
```

Comment:
This is wrong anyway. Per page has to be greater than 0, not equal to or greater than 0.